File: C/binary search/qwen-qwq-32b.c
#include <stdlib.h>

#include <time.h>

int compare(const void *a, const void *b) {

{

    int val1 = *(int*)a;

    int val2 = *(int*)b;

    if (val1 < val2) return -1;

    else if (val1 > val2) return 1;

    else return 0;

}

int binary_search(int arr[], int n, int target) {

{

    int low = 0;

    int high = n -1;

    while (low <= high) {

        int mid = low + (high - low)/2;

        if (arr[mid] == target) return mid;

        else if (target < arr[mid]) {

            high = mid -1;

        } else {

            low = mid +1;

    }

    return -1;

}

int main() {

    int arr[1000];

    srand(time(0));

    for (int i=0; i<1000; i++) {

        arr[i] = rand() % 10000; // random numbers between 0 and 9999.

    }

    qsort(arr, 1000, sizeof(int), compare);

    // Now test each element in the array.

    for (int i=0; i<1000; i++) {

        int target = arr[i];

        int index = binary_search(arr, 1000, target);

        if (index == -1 || arr[index] != target) {

            exit(EXIT_FAILURE);

        }

    }

    // Also, maybe test some non-present targets?

    // But the user might just need 1000 tests. The above loop does 1000 tests.

    return 0;

}

File: Python/binary search/qwen-qwq-32b.py
import random

def binary_search(arr, target):
    if not arr:
        return -1
    if target < arr[0] or target > arr[-1]:
        return -1
    low = 0
    high = len(arr) -1
    while low <= high:
        mid = (low + high) //2
        mid_val = arr[mid]
        if mid_val < target:
            low = mid +1
        elif mid_val > target:
            high = mid -1
        else:
            return mid
    return -1

import sys

def main():
    input_path = "inputs.txt" if len(sys.argv) < 2 else sys.argv[1]

    try:
        with open(input_path, 'r') as f:
            arr = [int(line.strip()) for line in f if line.strip()]
    except FileNotFoundError:
        return

    arr.sort()  # binary_search requires sorted array

    target = 15
    binary_search(arr, target)
if __name__ == "__main__":
    main()